### PR TITLE
Explain the new upcoming release train process

### PR DESCRIPTION
**Pull request by josvazg**
**Description of the change**

Add a description of the new upcoming release train process to produce releases going forward.

**Benefits**

This process sets a predictable release schedule and process that project consumers can rely on.

### CODE REVIEW

**Review by alvneiayu — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,7 +18,35 @@ When submitting a PR make sure that it:
...
+
+Releases happen monthly. A release train "leaves" on the 15th of each month, or the closest working date to that.
+ 
+#### Creation
+
+A release train is created by *branching* from `main` to `release/YYYYDDMM`, where `YYYYDDMM` is a ISO 8601 numbered timestamp of the release day date.
+
+#### Validation
+
+A `release/...` branch will go thought the release CI. If anything fails the release branch is dropped, the issue fixed in `main` and a new release train is started on a new branch.
```
Comment (alemorcuq) [inline CONTRIBUTING.md:40]:
```suggestion
The `release/...` branch will go through the release CI. If anything fails the release branch is dropped, the issue fixed in `main` and a new release train is started on a new branch.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -18,7 +18,35 @@ When submitting a PR make sure that it:
...
+
+### Tagging
```
Comment (alemorcuq) [inline CONTRIBUTING.md:42]: Should this be `####` ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -18,7 +18,35 @@ When submitting a PR make sure that it:
...
+
+Once the release passes all validations and is published, it is merged into `released`. Then, it is tagged with the final version, following SemVer semantics as `vX.Y.Z`.
+
+### Hot-fixing releases
```
Comment (alemorcuq) [inline CONTRIBUTING.md:46]: Should this be `####` ?

Comment (josvazg) [reply CONTRIBUTING.md:46]: yes! thanks!
```suggestion
#### Hot-fixing releases
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (josvazg) [reply CONTRIBUTING.md:42]: yes!
```suggestion
#### Tagging
```

**Review by alvneiayu — APPROVED** [review]
LGTM

File: CONTRIBUTING.md (hunk 3 above)
Comment (alemorcuq) [reply CONTRIBUTING.md:46]: I still see `###`

File: CONTRIBUTING.md (hunk 2 above)
Comment (alemorcuq) [reply CONTRIBUTING.md:42]: I still see `###`

**Review by alemorcuq — APPROVED** [review]

**Pull request merged**